pkg/store/proxy.go

```diff
@@ -49,12 +61,37 @@ type ProxyStore struct {
 	selectorLabels labels.Labels
 
 	responseTimeout time.Duration
+	metrics         *proxyStoreMetrics
+}
+
+func newProxyStoreMetrics(reg *prometheus.Registry) *proxyStoreMetrics {
+	var m proxyStoreMetrics
+
+	m.firstRecvDuration = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Name:       "thanos_proxy_first_recv_duration",
+		Help:       "Time to get first part data from store(ms).",
+		Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		MaxAge:     2 * time.Minute,
+	}, []string{"store", "payload"})
+
+	m.timeoutRecvCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_proxy_timeout_recv_count",
+		Help: "Timeout recv count.",
+	}, []string{"store"})
+
+	if reg != nil {
```

Comment:
Minor: what about create `Register`/`RegisterIn` method in `* proxyStoreMetrics ` struct?
```go
func(m *proxyStoreMetrics) RegisterIn(r *prometheus.Registry) {
    r.MustRegister(m.firstRecvDuration)
    r.MustRegister(m.timeoutRecvCount)
}
```

And usage will look like:
```go
m := newProxyStoreMetrics()
m.RegisterIn(reg)
```

So there will be no optional argument in func